src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -60,109 +134,76 @@ export class CalciteSwitch {
     }
   }
 
-  @Listen("click") onClick(e: MouseEvent): void {
+  @Listen("click")
+  onClick(e: MouseEvent): void {
     // prevent duplicate click events that occur
     // when the component is wrapped in a label and checkbox is clicked
     if (
-      (!this.disabled && this.el.closest("label") && e.target === this.inputProxy) ||
+      (!this.disabled && this.el.closest("label") && e.target === this.inputEl) ||
       (!this.el.closest("label") && e.target === this.el)
     ) {
-      this.updateSwitch(e);
+      this.updateSwitch();
     }
   }
 
-  @Listen("keydown") keyDownHandler(e: KeyboardEvent): void {
+  @Listen("keydown")
+  keyDownHandler(e: KeyboardEvent): void {
     const key = getKey(e.key);
     if (!this.disabled && (key === " " || key === "Enter")) {
-      this.updateSwitch(e);
+      this.updateSwitch();
     }
   }
 
-  @Watch("switched") switchWatcher(): void {
-    this.switched
-      ? this.inputProxy.setAttribute("checked", "")
-      : this.inputProxy.removeAttribute("checked");
-  }
-
-  private inputProxy: HTMLInputElement;
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
 
   connectedCallback(): void {
-    this.setupProxyInput();
-  }
-
-  disconnectedCallback(): void {
-    this.observer.disconnect();
+    this.guid = this.el.id || `calcite-switch-${guid()}`;
+    this.renderInput();
   }
 
-  componentWillRender(): void {
-    this.syncProxyInputToThis();
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private renderInput(): void {
+    this.inputEl = document.createElement("input");
```

Comment:
This input is created each time the `connectedCallback` is called which could be if the component is moved and would lead to multiple input elements being created, is that correct? I think it should only create and place the stored input once.